File: solutions/csharp/restaurantmanagementsystem/Payment.cs
namespace RestaurantManagementSystem
{
    public class Payment
    {
        public int Id { get; }
        public double Amount { get; }
        public PaymentMethod Method { get; }
        public PaymentStatus Status { get; }

        public Payment(int id, double amount, PaymentMethod method, PaymentStatus status)
        {
            Id = id;
            Amount = amount;
            Method = method;
            Status = status;
        }
    }
}

File: solutions/golang/restaurantmanagementsystem/payment.go
package restaurantmanagementsystem

type Payment struct {
	ID     int
	Amount float64
	Method PaymentMethod
	Status PaymentStatus
}

func NewPayment(id int, amount float64, method PaymentMethod, status PaymentStatus) *Payment {
	return &Payment{
		ID:     id,
		Amount: amount,
		Method: method,
		Status: status,
	}
}
